pkg/scaler/general.go

```diff
@@ -100,11 +100,20 @@ type GeneralController struct {
 	// Latest unstabilized recommendations for each autoscaler.
 	recommendations map[string][]timestampedRecommendation
 
+	// Multi goroutine read and write recommendations may unsafe.
+	recommendationsLock sync.RWMutex
+
 	// Latest autoscaler events
 	scaleUpEvents   map[string][]timestampedScaleEvent
 	scaleDownEvents map[string][]timestampedScaleEvent
 
+	// Multi goroutine read and write events may unsafe.
+	scaleUpEventsLock   sync.RWMutex
+	scaleDownEventsLock sync.RWMutex
+
```

Comment:
We use RWMutex， but I did not see where we call `RLock/RUnlock`?

Change to Mutex?

Reply:
Yes, you are right, I will revise it to sync.Mutex.